### PR TITLE
chore: upgrade docusaurus to 2.3.1 & react 18.2

### PR DESCRIPTION
**Pull request by blakef**
Upgrade these dependencies, so we can support deep-linking in [Tabs](https://deploy-preview-8225--docusaurus-2.netlify.app/docs/markdown-features/tabs/?current-os=ios#query-string).

### CODE REVIEW

**Comment by Simek** [comment]
Thanks for looking at Docusaurus upgrade! 👍 

There are several thing which need to be done,, beside bumping the version in `package.json`:
* we need to regenerate swizzled components, and re-apply our custom modifications, files are located in https://github.com/facebook/react-native-website/tree/main/website/src/theme, and there is a guide how to do this - https://docusaurus.io/docs/next/swizzling
* we need to verify if custom classes used in our SCSS files still exist, there is no other way than going through various pages and looking for regressions in comparison to the currently deployed PROD instance

Broken thing I have spotted on quick check:
* loading showcase page crashes Docusuarus - https://deploy-preview-3618--react-native.netlify.app/showcase
* announcement bar (at top) looks unstyled now

**Comment by blakef** [comment]
## Push for the Showcase issue coming soon.

I wrote some visual diff tooling to give me a better chance of catching these things.
CleanShot 2023-03-13 at 20 47 06@2x

**Comment by blakef** [comment]
@Simek I think this LGTM now.  When you have a moment would you mind looking over it again.  I've incorporated the swizzle changes and fixed the issue with showcase.

File: website/src/pages/showcase.js [hunk 1]
```diff
@@ -77,13 +77,13 @@ const Showcase = () => {
...
-  useEffect(() => {
-    setRandomizedApps(randomizeApps(others.filter(app => !app.pinned)));
-    setPinnedRandomizedApps(randomizeApps(others.filter(app => app.pinned)));
-  }, []);
+  const [pinnedRandomizedApps] = useState(
+    randomizeApps(others.filter(app => !app.pinned))
+  );
+  const [randomizedApps] = useState(
+    randomizeApps(others.filter(app => app.pinned))
+  );
```
Comment (Simek) [inline website/src/pages/showcase.js:86]: Removing `useEffect` wrap for entries randomization, leads to the rendering issue, where app labels do not match descriptions. This was written that way to address those issues.

 
Screenshot 2023-03-14 090331

**Comment by Simek** [comment]
Thank you for all the work put into the upgrade! 🙏 Besides the comment about Showcase issue, this LGTM! 👍 

Also, it looks like we have broken content container on versions page - https://deploy-preview-3618--react-native.netlify.app/versions - but surprisingly, this already seems to be an issue for PROD deployment, so we can address this in a separate PR.

**Comment by blakef** [comment]
I'm spamming netifly atm because the issue doesn't reproduce on my locally hosted version.  Apologies for the noise.

File: website/src/pages/showcase.js [hunk 2]
```diff
@@ -14,13 +14,15 @@ import ThemedImage from '@theme/ThemedImage';
...
 import IconExternalLink from '../theme/Icon/ExternalLink';
 
-const renderApp = (app, i) => {
-  const imgSource = !app.icon.startsWith('http')
-    ? useBaseUrl('img/showcase/' + app.icon)
-    : app.icon;
+const renderApp = (app, i) => <AppBox app={app} key={`app-${app.name}-${i}`} />;
+
+const AppBox = ({app}) => {
+  const imgSource = useBaseUrl(
```
Comment (blakef) [inline website/src/pages/showcase.js:20]: This was necessary to keep the # of hook calls consistent.  [`useBaseUrl` handles FQDNs](https://github.com/facebook/docusaurus/blob/2f75979bc5b4fd971225098e1c122efdd71d6e6f/packages/docusaurus/src/client/exports/__tests__/useBaseUrl.test.tsx#L41).

**Pull request merged**